File: programmers/lv1-두 개 뽑아서 더하기.py
def solution(numbers):
    n = len(numbers)
    ans = set()
    for i in range(n-1):
        for j in range(i+1, n):
            num = numbers[i] + numbers[j]
            if num not in ans:
                ans.add(num)
    return list(sorted(ans))
    

File: by-kotlin/kt-programmers/solutions/Lv1-두 개 뽑아서 더하기.kt
class Solution {
    fun solution(numbers: IntArray): List<Int> {
        val answer = mutableListOf<Int>()
        val n = numbers.size
        for (i in 0 until n - 1) {
            for (j in i + 1 until n) {
                answer.add(numbers[i] + numbers[j])
            }
        }

        return answer.distinct().sorted()
    }
}

fun main() {
    val s = Solution()
    print(s.solution(intArrayOf(2,1,3,4,1)))
}
